docs/kb/consistency.rst

```diff
@@ -0,0 +1,91 @@
+
+This is also related to the `CAP theorem <https://groups.google.com/d/forum/scylladb-users>`_, which states that in a distributed data system, 
+only two out of the following three guarantees can be satisfied:
+
+* **(Atomic) Consistency**: The same response is given to all identical requests.
+* **Availability**: Requests receive responses even during a partial system failure.
+* **Partition Tolerance**: Operations remain intact even when some nodes are unavailable.
+
+When discussing consistency in the context of the CAP theorem, the term typically applies to a single request/response operation sequence. 
+This is different from the meaning in the ACID database context, where consistency refers to transactions.
+
+ScyllaDB, as well as `Apache Cassandra <https://cassandra.apache.org/>`_ and other NoSQL databases, sacrifice a degree of consistency in order to 
+increase availability. Rather than providing strong consistency, they provide `eventual consistency <https://www.scylladb.com/glossary/eventual-consistency/>`_. This means that in some cases, a read request will fail to return the result of the latest WRITE. 
+
+In ScyllaDB (and Apache Cassandra), consistency is tunable - for a given query, the client can specify a `Consistency Level <https://university.scylladb.com/courses/scylla-essentials-overview/lessons/architecture/topic/consistency-level-cl/>`_.
+A Consistency Level is the number of replicas required to respond to a request for the request to be considered successful.  The possibility to
+specify a Consistency Level per query is known as `Tunable Consistency <https://docs.scylladb.com/stable/glossary.html#term-Tunable-Consistency>`_. 
+
+A database using the ACID model expects a unit of work to be atomic. It is all-or-nothing. While that unit of work is being done, the records or 
+tables affected may be locked. 
+
+On the other hand, distributed databases with a BASE model trade off strong consistency with high availability. The records stay available, 
+but once the query has been completed across a majority of `nodes <https://university.scylladb.com/courses/scylla-essentials-overview/lessons/architecture/topic/node/>`_, 
+or as defined by the Consistency Level, the transaction is deemed successful. Data replication across all nodes can take some time, but the data in all 
+the nodes will become consistent eventually. Once the query is deemed successful, queries for the data will consistently provide the updated data, 
+even before data replication reaches the last few nodes.
+
+Early results of eventual consistency data queries may not have the most recent updates. This is because it takes time for updates to reach replicas 
+across a distributed database cluster. In strong consistency, however, data is sent to every replica the moment a query is made. This causes delays 
+because responses to any new requests must wait while the replicas are updated. 
```

Comment:
I think this paragraph is somewhat misleading - it suggests that "strong consistency" needs CL=ALL (wait for all replicas to receive the write), but what about CL=QUORUM for both reads and writes, for example, don't we also consider that strong consistency?